src/PrettyPrinting.jl

```diff
@@ -1386,24 +1386,66 @@ end
 #
 ################################################################################
 
-# A non-compiletime preference
-function allow_unicode(flag::Bool)
-  old_flag = is_unicode_allowed()
-  @set_preferences!("unicode" => flag)
-  return old_flag
+@static if VERSION < v"1.8.0-DEV.1465"
+   ALLOW_UNICODE_OVERRIDE_VALUE = nothing
+else
+   ALLOW_UNICODE_OVERRIDE_VALUE::Union{Bool,Nothing} = nothing
 end
 
+@doc """
+    allow_unicode(flag::Bool; temporary::Bool=false) -> Bool
+
+Set whether unicode characters are allowed in pretty printing and returns the
+previous value.
+If `temporary` is `true`, then the change is only active for the current worker and session.
+Otherwise, the change is permanently saved in the preferences.
+A permanent change will always override a temporary change.
+
+This function may behave arbitrarily, if called from within an argument to
```

Comment:
```suggestion
This function may behave arbitrarily if called from within an argument to
```

Reply:
Actually thinking about this some more, I find "within an argument" a bit confusing. Maybe "within scope of a `with_unicode` do-block" -- that one can use other function beyond do-blocks is clear to the experts? Or we write "within scope of a function executed by `with_unicode`" ?

Reply:
reformulated